automation/run-tests-system.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash -ex
+
+EXEC_PATH=$(dirname "$(realpath "$0")")
+PROJECT_PATH="$(dirname $EXEC_PATH)"
+
+TEST_TYPE_ALL="all"
+TEST_TYPE_INTEG="integ"
+TEST_TYPE_INTEG_TIER1="integ_tier1"
+TEST_TYPE_INTEG_TIER2="integ_tier2"
+TEST_TYPE_INTEG_SLOW="integ_slow"
+
+PYTEST_OPTIONS="--verbose --verbose \
+        --log-level=DEBUG \
+        --log-date-format='%Y-%m-%d %H:%M:%S' \
+        --log-format='%(asctime)s %(filename)s:%(lineno)d %(levelname)s %(message)s' \
+        --durations=5 \
+        --cov /usr/lib/python*/site-packages/libnmstate \
+        --cov /usr/lib/python*/site-packages/nmstatectl \
+        --cov-report=term"
+
+function run_tests {
+    if [ $TEST_TYPE == $TEST_TYPE_ALL ] || \
+       [ $TEST_TYPE == $TEST_TYPE_INTEG ];then
+        pytest \
+          tests/integration \
+          --verbose --verbose \
+          ${nmstate_pytest_extra_args}
+    fi
+
+    if [ $TEST_TYPE == $TEST_TYPE_ALL ] || \
+       [ $TEST_TYPE == $TEST_TYPE_INTEG_TIER1 ];then
+        pytest \
+	  -m tier1 \
+          tests/integration \
+          --verbose --verbose \
+          --log-level=DEBUG \
+          ${nmstate_pytest_extra_args}
+    fi
+
+    if [ $TEST_TYPE == $TEST_TYPE_ALL ] || \
+       [ $TEST_TYPE == $TEST_TYPE_INTEG_TIER2 ];then
+        pytest \
+	  -m tier2 \
+          tests/integration \
+          --verbose --verbose \
+          ${nmstate_pytest_extra_args}
+    fi
+
+    if [ $TEST_TYPE == $TEST_TYPE_ALL ] || \
+       [ $TEST_TYPE == $TEST_TYPE_INTEG_SLOW ];then
+        pytest \
+	  -m slow --runslow \
+          tests/integration \
+          --verbose --verbose \
+          ${nmstate_pytest_extra_args}
+    fi
+}
+
+function write_separator {
+    local text="$(echo "${1}" | sed 's,., \0,g') "
+    local char="="
+
+    local textlength=$(echo -n "${text}" | wc --chars)
+    local cols="$(tput cols)"
+    local wraplength=$(((cols - textlength) / 2))
+
+    eval printf %.1s "${char}"'{1..'"${wraplength}"\}
+    echo -n "${text}"
+    wraplength=$((wraplength + ((cols - textlength) % 2)))
+    eval printf %.1s "${char}"'{1..'"${wraplength}"\}
+    echo
+}
+
+function run_exit {
+    write_separator "TEARDOWN"
+    dump_network_info
+}
+
+function modprobe_ovs {
+    lsmod | grep -q ^openvswith || modprobe openvswitch || { echo 1>&2 "Please run 'modprobe openvswitch' as root" exit 1; }
+}
+
+function check_iface_exist {
+    set +e
+    ip a s | grep -q $1
```

Comment:
The set setting should probably be restored after the call or not be touched.

Reply:
I will move that out. The `set +e` and `set -e` is needed because if grep does not find anything it will stop the whole run and we do not want that.

Reply:
Done, thanks!